drivers/periph_common/Kconfig.vbat

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2022 Otto-von-Guericke-Universität Magdeburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+if TEST_KCONFIG
+
+menuconfig MODULE_PERIPH_VBAT
+    bool "Backup Battery monitoring driver"
+    depends on HAS_PERIPH_VBAT
+    depends on HAS_PERIPH_ADC
+    select MODULE_PERIPH_ADC
+
+config MODULE_PERIPH_INIT_VBAT
+    bool "Auto initialize VBAT ADC line"
+    depends on MODULE_PERIPH_VBAT
+    default y if MODULE_PERIPH_INIT
+
+endif # TEST_KCONFIG
+
+menuconfig KCONFIG_USEMODULE_PERIPH_VBAT
+    bool "Configure backup battery monitoring peripheral driver"
+    depends on USEMODULE_PERIPH_VBAT
+    help
+        Configure backup battery monitoring peripheral driver using Kconfig.
+
+if KCONFIG_USEMODULE_PERIPH_VBAT
+
+config VBAT_ADC_VREF_MV
+    int "ADC reference voltage in mV"
+    default 3300
+    help
+        This is the reference voltage (VREF) of the ADC.
+        Often VREF is connected with VDDA, which is equal to VDD.
+
+endif # KCONFIG_USEMODULE_PERIPH_VBAT
```

Comment:
The reason it does not show up when `TEST_KCONFIG=1` is that, when that flag is active, Makefile dependency resolution does not run -> `USEMODULE_PERIPH_VBAT` is not defined. Although it implies some duplication, I'd go with the option of defining it again, in the first menuconfig. It could look something like:

```
menuconfig MODULE_PERIPH_VBAT
    bool "Backup Battery monitoring driver"
    depends on HAS_PERIPH_VBAT
    depends on HAS_PERIPH_ADC
    depends on TEST_KCONFIG
    select MODULE_PERIPH_ADC

if MODULE_PERIPH_VBAT

config MODULE_PERIPH_INIT_VBAT
    bool "Auto initialize VBAT ADC line"
    default y if MODULE_PERIPH_INIT

config VBAT_ADC_VREF_MV
    int "ADC reference voltage in mV"
    default 3300

endif # MODULE_PERIPH_VBAT

menuconfig KCONFIG_USEMODULE_PERIPH_VBAT
    bool "Configure backup battery monitoring peripheral driver"
    depends on USEMODULE_PERIPH_VBAT
    help
        Configure backup battery monitoring peripheral driver using Kconfig.

config VBAT_ADC_VREF_MV
    int "ADC reference voltage in mV"
    default 3300
    depends on KCONFIG_USEMODULE_PERIPH_VBAT
    help
        This is the reference voltage (VREF) of the ADC.
        Often VREF is connected with VDDA, which is equal to VDD.
```

Reply:
Thank you. I am going to squash that in right away.